### PR TITLE
Initial boxplot code

### PR DESCRIPTION
**Pull request by Turgar**
Initial pull request for a boxplot. It needs a few updates (like changing the median line and fixing pop-ups).

### CODE REVIEW

**Comment by robinfhu** [comment]
Looking at it...

**Comment by liquidpele** [comment]
Neat!  Do you have time to add a simple unit test that just loads the chart to ensure it's not completely broken?

**Comment by robinfhu** [comment]
Can you also add a page to the examples/ folder

File: src/models/boxPlot.js [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        y:       {get: function(){return getY;}, set: function(_){getY=_;}},
+        xScale:  {get: function(){return x;}, set: function(_){x=_;}},
+        yScale:  {get: function(){return y;}, set: function(_){y=_;}},
+        xDomain: {get: function(){return xDomain;}, set: function(_){xDomain=_;}},
+        yDomain: {get: function(){return yDomain;}, set: function(_){yDomain=_;}},
+        xRange:  {get: function(){return xRange;}, set: function(_){xRange=_;}},
+        yRange:  {get: function(){return yRange;}, set: function(_){yRange=_;}},
+        valueFormat:    {get: function(){return valueFormat;}, set: function(_){valueFormat=_;}},
+        id:          {get: function(){return id;}, set: function(_){id=_;}},
+        // rectClass: {get: function(){return rectClass;}, set: function(_){rectClass=_;}},
```
Comment (robinfhu) [inline src/models/boxPlot.js]: Remove commented code

File: src/models/boxPlotChart.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    yAxis
+        .orient((rightAlignYAxis) ? 'right' : 'left')
+        .tickFormat(d3.format(',.1f'))
+    ;
+    
+    tooltip
+        .duration(0)
+        .contentGenerator(function(d) {
+            return '<h3>' + d.data.label + '</h3>' + 
+                '<table style="margin-top: 0px; margin-bottom: 0px; margin-right: 0px; width: 100%; text-align: center;">' +
```
Comment (robinfhu) [inline src/models/boxPlotChart.js]: I don't think it's necessary to put inline styles here.  Let the user style it.

**Comment by liquidpele** [comment]
Ah yea, just make a demo page in the examples folder and link to it from site.html :)

File: src/models/boxPlotChart.js (hunk 2 above)
Comment (robinfhu) [inline src/models/boxPlotChart.js]: Or if you want, add some styles to the CSS file

**Comment by robinfhu** [comment]
@Turgar any updates?

**Comment by Turgar** [comment]
Hoping to make the changes this weekend. It's been a busy week.

**Comment by liquidpele** [comment]
No worries, thanks for the update!

**Comment by Turgar** [comment]
How do you normally unit test? I see .coffee scripts and various html pages.

Thanks.

**Comment by robinfhu** [comment]
@Turgar  look at the .coffee files in the test/ folder.  If you look at examples it should make sense.  You basically generate a chart in a browser and run basic checks on it.

**Comment by liquidpele** [comment]
https://github.com/novus/nvd3/tree/development#tips-for-testing

just running 'grunt' should run the karma unit test framework that uses the coffee scripts in the test directory.

**Comment by Turgar** [comment]
Thanks for the docs. I've finished the examples and code clean-up, but I
still can't quite get the unit tests working.

grunt production works OK, but following with grunt shows this error:

Running "karma:unit" (karma) task
Firefox 36.0.0 (Mac OS X 10.9) ERROR
  ReferenceError: d3 is not defined
  at /Users/turg/git/Turgar.github.io/src/core.js:9

A Google search mentions a karma config, but I don't see one included with
the project. Bear with me, I've never used node, etc... before today.

Thanks

**Comment by liquidpele** [comment]
Yea, looks like karma is expecting d3 to be there since that's a dependency...   did you do "npm install" to get the dependencies before running grunt?

**Comment by Turgar** [comment]
Ah, got it. npm install has the warning below. I thought I had already
tried installing d3, but running 'npm install d3' fixes the issue.

npm WARN peerDependencies The peer dependency d3 included from nvd3 will no
npm WARN peerDependencies longer be automatically installed to fulfill the
peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend
on it explicitly.

I should be able to wrap things up by Sunday. Sorry about the delays.

Thanks

**Comment by Turgar** [comment]
Can you take a look at the latest? I've added an example and unit tests, and made some additional changes to the code.

I committed my local build too... not sure if that was a good idea.

Thanks,
Ryan

**Comment by liquidpele** [comment]
I can take a look, but please don't include anything under the build directory in your pull request, it makes it a lot harder to code review and it just gets overwritten later :)

**Comment by liquidpele** [comment]
Yup, this looks good to me, just need to remove the files under the "build" directory and I'll merge it in :)

**Comment by Turgar** [comment]
Awesome, thanks! I've reverted the build files to the previous commit --
they should be restored to your branch now.

**Comment by liquidpele** [comment]
awesome thank you for all the work!

**Comment by liquidpele** [comment]
@Turgar just FYI, I noticed 2 things looking at the example:

1. the "sample B" one seems to get cut off at the top, I assume it's not taking the whiskers or outliers into account when it calculates the Y axis range? 
2. You can't get the whisker values on hover anywhere, at least that I can see... 

https://nvd3-community.github.io/nvd3/examples/boxPlot.html

**Comment by Turgar** [comment]
Hey @liquidpele,

1. The Sample B data actually has Q3 == top whisker == 400 with no high
outliers. I probably wanted to see what that edge case looked like, but it
makes an odd example. I should make the whiskers optional anyway.
2. I didn't implement that. Shouldn't be hard though, let me take a look.

Thanks.

**Comment by spsaucier** [comment]
Is this still in development? It would be helpful, and it looks like it's nearly there.

**Comment by liquidpele** [comment]
It's been merged for 2 versions already...

**Pull request merged**